client/src/components/CartIcon.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useContext, useEffect } from "react";
+import { Badge, IconButton } from "@material-ui/core";
+import { withStyles, makeStyles } from "@material-ui/core/styles";
+import ShoppingCartIcon from "@material-ui/icons/ShoppingCart";
+import DialogControl from "./Dialogs/DialogControl";
+import { UserContext, UserDispatchContext } from "../context/UserContext";
+import { getCartInfo } from "../actions/cartActions";
+
+const StyledBadge = withStyles((theme) => ({
+  badge: {
+    right: -3,
+    top: 13,
+    border: `2px solid ${theme.palette.background.paper}`,
+    padding: "0 4px",
+  },
+}))(Badge);
```

Comment:
It would be better to be consistent and style this the same way as other components in the makeStyles function